### PR TITLE
Update OSM image for e2e tests

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
Using the image against latest commit hash to include the containerd 1.6 changes.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: examples/operating-system-manager.yaml [hunk 1]
```diff
@@ -1076,7 +1076,8 @@ spec:
     spec:
       serviceAccountName: operating-system-manager-webhook
       containers:
-        - image: quay.io/kubermatic/operating-system-manager:v1.0.0
+        # TODO: Update this to a semver tag before release.
+        - image: quay.io/kubermatic/operating-system-manager:ea2250874bbcea46ad956cbba8972be7ffdf9ce2
```
Comment (kron4eg) [inline examples/operating-system-manager.yaml:1080]: how about releasing new OSM version instead?

Comment (ahmedwaleedmalik) [reply examples/operating-system-manager.yaml:1080]: There are few more things that I'd like to add in OSM before the next release.

**Pull request merged**